### PR TITLE
Add PostgreSQLConnectorConfig to expose SSL configuration options

### PR DESCRIPTION
**Pull request by rpless**
This PR exposes the SSL settings so a user can enable ssl if they are using postgres as their metadata storage.

I also noticed that the version of the postgres driver is quite old. The latest version is `42.2.4.jre7`. Should I bump up it up to that either here or in a another PR?

### CODE REVIEW

**Comment by rpless** [comment]
At some point I switch from the Druid intellij format back to my defaults and missed all the checkstyle warnings. Fixing them. Sorry about that.

File: docs/content/development/extensions-core/postgresql.md [hunk 1]
```diff
@@ -47,3 +47,19 @@ Make sure to [include](../../operations/including-extensions.html) `postgresql-m
...
   ```
+
+## Configuration
+In most cases, the configuration options map directly to the [postgres jdbc connection options](https://jdbc.postgresql.org/documentation/head/connect.html).
+
+|Property|Description|Default|Required|
+|--------|-----------|-------|--------|
+| useSSL | Enables SSL | `false` | no |
+| sslPassword | The [Password Provider](../../operations/password-provider.html) or String password for the client's key. | none | no |
+| sslFactory | The class name to as as the `SSLSocketFactory` | none | no |
```
Comment (jon-wei) [inline docs/content/development/extensions-core/postgresql.md]: typo: "to as as"

File: docs/content/development/extensions-core/postgresql.md [hunk 2]
```diff
@@ -47,3 +47,19 @@ Make sure to [include](../../operations/including-extensions.html) `postgresql-m
   druid.metadata.storage.connector.user=druid
   druid.metadata.storage.connector.password=diurd
   ```
+
+## Configuration
+In most cases, the configuration options map directly to the [postgres jdbc connection options](https://jdbc.postgresql.org/documentation/head/connect.html).
+
+|Property|Description|Default|Required|
+|--------|-----------|-------|--------|
+| useSSL | Enables SSL | `false` | no |
```
Comment (jon-wei) [inline docs/content/development/extensions-core/postgresql.md]: The druid docs generally put the full property name in these tables, e.g. `druid.metadata.postgres.ssl.useSSL`, can you change these to that form?

**Comment by jon-wei** [comment]
@rpless Thanks for the contribution!

> I also noticed that the version of the postgres driver is quite old. The latest version is 42.2.4.jre7. Should I bump up it up to that either here or in a another PR?

Let's do that in a separate PR

**Comment by rpless** [comment]
Thanks for the feedback @jon-wei! I've updated the PR with it.

> Let's do that in a separate PR
Sounds good.

File: docs/content/development/extensions-core/postgresql.md [hunk 3]
```diff
@@ -47,3 +47,19 @@ Make sure to [include](../../operations/including-extensions.html) `postgresql-m
...
   ```
+
+## Configuration
+In most cases, the configuration options map directly to the [postgres jdbc connection options](https://jdbc.postgresql.org/documentation/head/connect.html).
+
+|Property|Description|Default|Required|
+|--------|-----------|-------|--------|
+| `druid.metadata.postgres.useSSL` | Enables SSL | `false` | no |
+| `druid.metadata.postgres.sslPassword` | The [Password Provider](../../operations/password-provider.html) or String password for the client's key. | none | no |
+| `druid.metadata.postgres.sslFactory` | The class name to as the `SSLSocketFactory` | none | no |
```
Comment (jon-wei) [inline docs/content/development/extensions-core/postgresql.md]: Should probably be "The class name to use as"

Comment (rpless) [reply docs/content/development/extensions-core/postgresql.md]: Yep good catch. I'll change that.

**Pull request merged**